client/interactive.js

```diff
@@ -9,42 +9,40 @@ export class FormulaBunBotInteracive extends FormulaBunBotBase {
     super(param);
     this.players = {};
     this.server = {};
-    this.on("ready", () => {
-      this.registercommands();
-      this.ws.on("INTERACTION_CREATE", (interaction) =>
-        this.respond(interaction)
-      );
+    this.on('ready', async () => {
+      await this.registercommands();
+      this.on('interactionCreate', (interaction) => {
+        if (interaction.isCommand()) {
+          this.respond(interaction);
+        }
+      });
     });
   }
 
-  registercommands() {
+  async registercommands() {
     for (let c in commands) {
       if (commands.hasOwnProperty(c)) {
-        let guild,
-          application = this.api.applications(this.user.id);
-
+        let commandsObject;
         switch (process.env.DISCORDBOT_ENV) {
-          case "test":
-            guild = application.guilds(TEST_GUILD);
+          case 'test':
+            commandsObject = this.application.commands;
             break;
-          case "deploy":
-            guild = application;
+          case 'deploy':
+            commandsObject = (await this.guilds.fetch(TEST_GUILD)).commands;
             break;
```

Comment:
I'm pretty sure you swapped test and deploy around. Adding a command to a single guild is much faster than creating an application command. I use the test guild for, well, testing.

Reply:
I did accidentally swap them lamo.